cdc/sinkv2/metrics/kafka/collector.go

```diff
@@ -0,0 +1,282 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tiflow/cdc/model"
+	"github.com/pingcap/tiflow/pkg/kafka"
+	"github.com/pingcap/tiflow/pkg/util"
+	"github.com/rcrowley/go-metrics"
+	"go.uber.org/zap"
+)
+
+// flushMetricsInterval specifies the interval of refresh sarama metrics.
+const flushMetricsInterval = 5 * time.Second
+
+// Sarama metrics names, see https://pkg.go.dev/github.com/Shopify/sarama#pkg-overview.
+const (
+	// metrics at producer level.
+	batchSizeMetricName        = "batch-size"
+	recordSendRateMetricName   = "record-send-rate"
+	recordPerRequestMetricName = "records-per-request"
+	compressionRatioMetricName = "compression-ratio"
+
+	// metrics at broker level.
+	incomingByteRateMetricNamePrefix   = "incoming-byte-rate-for-broker-"
+	outgoingByteRateMetricNamePrefix   = "outgoing-byte-rate-for-broker-"
+	requestRateMetricNamePrefix        = "request-rate-for-broker-"
+	requestSizeMetricNamePrefix        = "request-size-for-broker-"
+	requestLatencyInMsMetricNamePrefix = "request-latency-in-ms-for-broker-"
+	requestsInFlightMetricNamePrefix   = "requests-in-flight-for-broker-"
+	responseRateMetricNamePrefix       = "response-rate-for-broker-"
+	responseSizeMetricNamePrefix       = "response-size-for-broker-"
```

Comment:
We can add another metric to track the amount of topics created for each changefeed in TiCDC.